src/dataflow-types/src/client.rs

```diff
@@ -176,6 +176,125 @@ pub enum ComputeCommand<T = mz_repr::Timestamp> {
+                            id: Some(id.into()),
+                            frontier: Some(frontier.into()),
+                        })
+                        .collect(),
+                }),
+                ComputeCommand::Peek(peek) => Peek(peek.into()),
+                ComputeCommand::CancelPeeks { uuids } => CancelPeeks(ProtoCancelPeeks {
+                    uuids: uuids.into_iter().map(|id| id.into_proto()).collect(),
+                }),
+            }),
+        }
+    }
+}
+
+impl TryFrom<ProtoComputeCommand> for ComputeCommand<mz_repr::Timestamp> {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoComputeCommand) -> Result<Self, Self::Error> {
+        use proto_compute_command::Kind::*;
+        use proto_compute_command::*;
+        match x.kind {
+            Some(CreateInstance(ProtoCreateInstance { logging })) => Ok(
+                ComputeCommand::CreateInstance(logging.map(TryInto::try_into).transpose()?),
+            ),
+            Some(DropInstance(())) => Ok(ComputeCommand::DropInstance),
+            Some(CreateDataflows(ProtoCreateDataflows { dataflows })) => {
+                Ok(ComputeCommand::CreateDataflows(
+                    dataflows
+                        .into_iter()
+                        .map(TryInto::try_into)
+                        .collect::<Result<Vec<_>, _>>()?,
+                ))
+            }
+            Some(AllowCompaction(ProtoCompactions { collections })) => {
+                Ok(ComputeCommand::AllowCompaction(
+                    collections
+                        .into_iter()
+                        .map(|collection| {
+                            Ok((
+                                collection.id.try_into_if_some("ProtoCompaction::id")?,
+                                collection.frontier.map(Into::into).ok_or_else(|| {
+                                    TryFromProtoError::missing_field("ProtoCompaction::frontier")
+                                })?,
+                            ))
+                        })
+                        .collect::<Result<Vec<_>, TryFromProtoError>>()?,
+                ))
+            }
+            Some(Peek(peek)) => Ok(ComputeCommand::Peek(peek.try_into()?)),
+            Some(CancelPeeks(ProtoCancelPeeks { uuids })) => Ok(ComputeCommand::CancelPeeks {
+                uuids: uuids
+                    .into_iter()
+                    .map(Uuid::from_proto)
+                    .collect::<Result<BTreeSet<_>, _>>()?,
+            }),
+            None => Err(TryFromProtoError::missing_field(
+                "ProtoComputeCommand::kind",
+            )),
+        }
+    }
+}
+
+impl Arbitrary for ComputeCommand<mz_repr::Timestamp> {
+    type Strategy = BoxedStrategy<Self>;
+    type Parameters = ();
+
+    fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+        prop_oneof![
+            any::<Option<LoggingConfig>>()
+                .prop_map(|logging| ComputeCommand::CreateInstance(logging)),
```

Comment:
[Eta-reduce](https://en.wikipedia.org/wiki/Lambda_calculus#%CE%B7-reduction) this (required by the linter): 

```rust
prop_map(ComputeCommand::CreateInstance)
```